target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity

```diff
@@ -4,8 +4,14 @@
 
 get_device_irq() {
 	local device="$1"
-
-	local line=$(grep -m 1 "${device}\$" /proc/interrupts)
+	local line
+	local seconds
+
+	# wait up to 10 seconds for the irq/device to appear
+	for seconds in 1 2 3 4 5 6 7 8 9 10; do
+		line=$(grep -m 1 "${device}\$" /proc/interrupts) && break
+		sleep 1
+	done
```

Comment:
can you change this to a pool logic and sleep for 2 second ?

a pool logic = a while loop... the counter increase and the while terminate when a value is reached.

Reply:
Thanks for the comment, but can you explain to me the reason for this change?

Reply:
better readable code, just that. Also sleeping for 2 second IMHO should make this function sleep less in most of the case.